### PR TITLE
Video preview

### LINKED ISSUE #17: Add video player support for viewing response of video mimetype
**Use case**

User should be able to directly play video in the response pane when the response is of content type video/H264, video/H265, video/mp4 etc. Full list of mimetypes available [here](https://www.iana.org/assignments/media-types/media-types.xhtml#video). Currently, the app allows users to download the video response.

### PR DESCRIPTION
**Pull request by mdrazak2001**
## Video Preview

A video preview feature addition 

## Related Issues

- Related Issue #17
- Closes #17

### Checklist
- [x] I have gone through the [contributing guide](https://github.com/foss42/apidash/blob/main/CONTRIBUTING.md)
- [x] I have run the tests (`flutter test`) and all tests are passing.

## Added/updated tests?
_We encourage you to add relevant test cases._

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests have not been included_
- [x] I need help with writing tests

### CODE REVIEW

**Comment by mdrazak2001** [comment]
Here is the proposed video preview feature:

https://github.com/foss42/apidash/assets/60781022/0f72b512-bbf0-4675-85af-a97dc3ec978b

**Comment by opxdelwin** [comment]
@mdrazak2001 #219  is being tracked at https://github.com/foss42/apidash/pull/229

**Comment by ashitaprasad** [comment]
@mdrazak2001 Some issues - the play/pause button is always present on the screen, which should not be the case. Also, there is a gray background which does not go with the API Dash theme.
Also, do look at tests for other response previewers like PDF, image etc. and you shall be able to figure out how to write tests.

**Comment by mdrazak2001** [comment]
@ashitaprasad thanks for the feedback! Will work on these issues and get back. Also am I supposed to raise a different PR for a potential fix for #219 ?

**Comment by ashitaprasad** [comment]
@mdrazak2001 Yes, raise a different PR for https://github.com/foss42/apidash/issues/219

**Comment by mdrazak2001** [comment]
@ashitaprasad - Have resolved all reported issues:
1. play/pause button is being hidden when you stay stagnant or don't hover over the player
2. gray background disappears when you don't hover over the player
3. Integration Tests added for the same

Updated Demo:

https://github.com/foss42/apidash/assets/60781022/a17ffa91-d2fd-450a-8f5a-5118c514dd94

**Comment by ashitaprasad** [comment]
@mdrazak2001 Can you remove the gray overlay and just show the paly/pause button when the user hovers on the video. The gray overlay is not going well with the theme of API Dash.

File: lib/widgets/response_widgets.dart [hunk 1]
```diff
@@ -351,13 +352,15 @@ class BodySuccess extends StatefulWidget {
       required this.options,
       required this.bytes,
       this.formattedBody,
-      this.highlightLanguage});
+      this.highlightLanguage,
+      this.videoUrl});
```
Comment (animator) [inline lib/widgets/response_widgets.dart:356]: You cannot use videoUrl, the video has to be played by decoding the bytes which is provided in the response.

**Comment by ashitaprasad** [comment]
@mdrazak2001 Can you please raise a new PR with the right approach where the bytes received as content are written to a temporary file that is input to [VideoPlayerController.file](https://pub.dev/documentation/video_player/latest/video_player/VideoPlayerController/VideoPlayerController.file.html) instead of network url.

**Comment by mdrazak2001** [comment]
Hey @ashitaprasad  @animator , thanks for the feedback really appreciate it, was a bit occupied in the past week sorry for the delayed response. Sure I will research how to formulate this approach in flutter env and get back to you by next week.

**Comment by ashitaprasad** [comment]
Sure @mdrazak2001 👍

**Pull request closed without merging**